config/routes.rb

```diff
@@ -2,5 +2,11 @@
   devise_for :users
   root to: 'pages#home'
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-  resources :sites, only: [:index, :show, :new, :create, :edit, :destroy]
+  resources :sites, only: [:index, :show, :new, :create, :edit, :destroy] do
+    member do
```

Comment:
I guess this works, but aren't we missing `resources :contents`, nested into sites between the  `resources :sites` and `member do` ? Videos is a member of contents and not sites, right?